## Using Pandas

In [15]:
print("Student Name: {sn} \nStudent Number: {num}".format(sn="Andrianna Wardill", num="041067235"))

Student Name: Andrianna Wardill 
Student Number: 041067235


**Data Dictionary**

* Name - Name of the person
* Net worth in billions - Net worth of the person in billions
* Country - Country where the person is from
* Source - Source of the income
* Rank - Rank of the person in the Billionaires list
* Age - Age of the person
* Industry - Industry to which the person is related to

### Importing Pandas Library

In [2]:
import pandas as pd

### Loading the dataset in a pandas dataframe

In [3]:
df = pd.read_csv('Forbes-list-2020.csv')

### Print top 5 rows 

In [4]:
print('Umer Altaf')
df.head(5)

Umer Altaf


,Name,NetWorth in billions,Country,Source,Rank,Age,Industry
0,Jeff Bezos,177.0,United States,Amazon,1,57,Technology
1,Elon Musk,151.0,United States,"Tesla, SpaceX",2,49,Automotive
2,Bernard Arnault & family,150.0,France,LVMH,3,72,Fashion & Retail
3,Bill Gates,124.0,United States,Microsoft,4,65,Technology
4,Mark Zuckerberg,97.0,United States,Facebook,5,36,Technology


#### <span style="color:crimson">Q1. Whose name is assigned to the 0th index in the dataframe</span>



In [5]:
df.loc[0,]

Name                       Jeff Bezos
NetWorth in billions            177.0
Country                 United States
Source                         Amazon
Rank                                1
Age                                57
Industry                   Technology
Name: 0, dtype: object

### Print the dimensions of data

In [4]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2755 entries, 0 to 2754
Data columns (total 7 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   Name                  2755 non-null   object 
 1   NetWorth in billions  2755 non-null   float64
 2   Country               2755 non-null   object 
 3   Source                2755 non-null   object 
 4   Rank                  2755 non-null   int64  
 5   Age                   2755 non-null   int64  
 6   Industry              2755 non-null   object 
dtypes: float64(1), int64(2), object(4)
memory usage: 150.8+ KB


#### <span style="color:crimson">Q2. How many records do you have in this dataset?</span>

There are 2755 records.

#### <span style="color:crimson">Q3. What is the data type of the variable Source</span>

Source is of date type object

### Checking for missing values

In [7]:
df.isna().sum()

Name                    0
NetWorth in billions    0
Country                 0
Source                  0
Rank                    0
Age                     0
Industry                0
dtype: int64

#### <span style="color:crimson">Q4. How many missing values exists in the industry feature</span>

There are no missing values in the Industry feature.

### Print the basic stats for this dataset

In [12]:
df.describe()

,NetWorth in billions,Rank,Age
count,2755.000000,2755.000000,2755.000000
mean,4.749220,1345.663521,63.014519
std,9.615358,772.669811,13.259017
min,1.000000,1.000000,18.000000
25%,1.500000,680.000000,54.000000
50%,2.300000,1362.000000,62.000000
75%,4.200000,2035.000000,72.000000
max,177.000000,2674.000000,97.000000


#### <span style="color:crimson">Q5. Who is the yongest billionare in this list?</span>

In [12]:
# Using iloc
youngest = df.sort_values('Age').iloc[0]['Name']
print(youngest)

Kevin David Lehmann


#### <span style="color:crimson">Q6. What is the output of `df.iloc[1200,3]`?</span>

In [16]:
output = df.iloc[1200,3]
print(output)

real estate


### Create a new column `Year_of_birth`
Note: The dataset is from the year 2020. The age of those Billionaires is as of 2020.

In [20]:
# create a new column 'Year_of_birth'
current_year = 2020
df['Year_of_birth'] = current_year - df['Age']

In [21]:
#view the data
df.head(5)

,Name,NetWorth in billions,Country,Source,Rank,Age,Industry,Year_of_birth
0,Jeff Bezos,177.0,United States,Amazon,1,57,Technology,1963
1,Elon Musk,151.0,United States,"Tesla, SpaceX",2,49,Automotive,1971
2,Bernard Arnault & family,150.0,France,LVMH,3,72,Fashion & Retail,1948
3,Bill Gates,124.0,United States,Microsoft,4,65,Technology,1955
4,Mark Zuckerberg,97.0,United States,Facebook,5,36,Technology,1984


### Select Billionaires born in 1996


In [25]:
# Using where condition
born_1996 = df[df['Year_of_birth'] == 1996]
print(born_1996)

                    Name  NetWorth in billions Country       Source  Rank  \
2122         Wang Zelong                   1.5   China    chemicals  2035   
2143  Alexandra Andresen                   1.4  Norway  investments  2141   

      Age         Industry  Year_of_birth  
2122   24  Metals & Mining           1996  
2143   24      Diversified           1996  


#### <span style="color:crimson">Q7. How many Billionaires were born in 1996?</span>

In [26]:
born_1996.count()

Name                    2
NetWorth in billions    2
Country                 2
Source                  2
Rank                    2
Age                     2
Industry                2
Year_of_birth           2
dtype: int64